AK/StringView.cpp

```diff
@@ -75,6 +75,34 @@ Vector<StringView> StringView::split_view(const char separator, bool keep_empty)
     return v;
 }
 
+Vector<StringView> StringView::split_view(const StringView& separator, bool keep_empty) const
+{
+    if (is_empty())
+        return {};
+
+    StringView view { *this };
+
+    Vector<StringView> parts;
+    if (separator.is_empty()) {
```

Comment:
Mmm, it's not clear to me how that should work. I'd rather have this method `ASSERT(!separator.is_empty())`, and have the callers deal with the empty case. Unless you can make an argument why it's a good idea to handle it this way.

Reply:
Not really, no.
we could also go the javascript way and return the characters as strings?

Reply:
Yeah, let's not do either of those. If the caller is passing an empty string it's probably a mistake on their side. Haven't validated input or something. So it's a good idea to assert to catch those.

Reply:
Let's assert then.
fine by me.